viogpu/viogpu3d/viogpu3d.vcxproj

```diff
@@ -0,0 +1,218 @@
+      <PreprocessorDefinitions>VIOGPU_3D=1;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <WppEnabled>true</WppEnabled>
+      <WppScanConfigurationData>trace.h</WppScanConfigurationData>
+    </ClCompile>
+    <Link>
+      <AdditionalDependencies>%(AdditionalDependencies);$(KernelBufferOverflowLib);$(DDK_LIB_PATH)ntoskrnl.lib;$(DDK_LIB_PATH)hal.lib;$(DDK_LIB_PATH)wmilib.lib;$(DDK_LIB_PATH)\displib.lib;virtiolib.lib</AdditionalDependencies>
+      <AdditionalLibraryDirectories>..\..\VirtIO\$(OutDir);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+    </Link>
+    <PostBuildEvent>
+    </PostBuildEvent>
+  </ItemDefinitionGroup>
+  <ItemGroup>
+    <ClInclude Include="..\common\baseobj.h" />
+    <ClInclude Include="..\common\bitops.h" />
+    <ClInclude Include="..\common\helper.h" />
+    <ClInclude Include="..\common\viogpu_queue.h" />
+    <ClInclude Include="driver.h" />
+    <ClInclude Include="viogpu_command.h" />
+    <ClInclude Include="viogpu_device.h" />
+    <ClInclude Include="trace.h" />
+    <ClInclude Include="viogpu_allocation.h" />
+    <ClInclude Include="viogpu_adapter.h" />
+    <ClInclude Include="resource.h" />
+    <ClInclude Include="viogpu_vidpn.h" />
+  </ItemGroup>
+  <ItemGroup>
+    <ClCompile Include="..\common\viogpu_idr.cpp" />
+    <ClCompile Include="..\common\viogpu_pci.cpp" />
+    <ClCompile Include="..\common\viogpu_queue.cpp" />
+    <ClCompile Include="driver.cpp" />
+    <ClCompile Include="viogpu_command.cpp" />
+    <ClCompile Include="viogpu_device.cpp" />
+    <ClCompile Include="viogpu_allocation.cpp" />
+    <ClCompile Include="viogpu_adapter.cpp" />
+    <ClCompile Include="..\common\baseobj.cpp" />
+    <ClCompile Include="..\common\bitops.cpp" />
+    <ClCompile Include="viogpu_vidpn.cpp" />
+  </ItemGroup>
+  <ItemGroup>
+    <ResourceCompile Include="viogpu3d.rc" />
+  </ItemGroup>
+  <ItemGroup>
+    <FilesToPackage Include="$(OutDir)viogpu3d.sys">
+      <PackageRelativeDirectory>
+      </PackageRelativeDirectory>
+    </FilesToPackage>
+  </ItemGroup>
+
+  <ItemGroup Condition="Exists('$(MESA_PREFIX)\bin\viogpu_d3d10.dll')">
+    <FilesToPackage Include="$(MESA_PREFIX)\bin\viogpu_d3d10.dll">
+      <PackageRelativeDirectory>
+      </PackageRelativeDirectory>
+    </FilesToPackage>
+    <FilesToPackage Include="$(MESA_PREFIX)\bin\viogpu_wgl.dll">
+      <PackageRelativeDirectory>
+      </PackageRelativeDirectory>
+    </FilesToPackage>
+    <FilesToPackage Include="$(MESA_PREFIX)\bin\z.dll">
+      <PackageRelativeDirectory>
+      </PackageRelativeDirectory>
+    </FilesToPackage>
+  </ItemGroup>
+
+  <ItemGroup Condition="Exists('$(MESA_PREFIX)\bin\viogpu_d3d10.dll')">
+    <Inf Include="viogpu3d.inx" />
+  </ItemGroup>
+  <!--  <ItemGroup Condition="'$(BUILD_RESOLUTIONS_USERMODE)'!=''"> -->
+  <ItemGroup Condition="Exists('$(MESA_PREFIX)\bin\viogpu_d3d10.dll')">
```

Comment:
This was added to avoid failing CI and to simplify building other drivers without compiling mesa (which is quite heavy). Maybe warning will suffice for now?

Reply:
Currently, this project is not in root solution, so CI will not touch it.  

In any case, building KMD and UMD independently is not a good idea. When I will create a CI for the 3D driver I will add mesa as a submodule and it will be a dependency for KMD. I am ok to remove this condition when adding the UMD project and keep it for now.